### PR TITLE
[WFLY-6787] WebSockets compression should be allowed by default

### PR DESCRIPTION
**Pull request by iweiss**
Issue: https://issues.jboss.org/browse/WFLY-6787

### CODE REVIEW

File: undertow/src/main/java/org/wildfly/extension/undertow/deployment/UndertowDeploymentInfoService.java [hunk 1]
```diff
@@ -982,6 +983,10 @@ private DeploymentInfo createServletConfig() throws StartException {
                 webSocketDeploymentInfo.setBuffers(servletContainer.getWebsocketsBufferPool().getValue());
                 webSocketDeploymentInfo.setWorker(servletContainer.getWebsocketsWorker().getValue());
                 webSocketDeploymentInfo.setDispatchToWorkerThread(servletContainer.isDispatchWebsocketInvocationToWorker());
+
+                // Enables compression. This fixes WFLY-6787
+                webSocketDeploymentInfo.addExtension(new PerMessageDeflateHandshake(false));
```
Comment (stuartwdouglas) [inline undertow/src/main/java/org/wildfly/extension/undertow/deployment/UndertowDeploymentInfoService.java]: This really needs a flag to make it configurable

Comment (iweiss) [inline undertow/src/main/java/org/wildfly/extension/undertow/deployment/UndertowDeploymentInfoService.java]: I agree, but the way it is now is a regression. I opened a JIRA to have this properly configured, https://issues.jboss.org/browse/WFLY-6784.

**Comment by bstansberry** [comment]
I'm putting a 10.x label on this just so it stays on the radar for 10.x as we try and filter PRs. @stuartwdouglas if you think it's wrong for 10.x, please change the label to 11.x.

File: undertow/src/main/java/org/wildfly/extension/undertow/deployment/UndertowDeploymentInfoService.java (hunk 1 above)
Comment (bstansberry) [inline undertow/src/main/java/org/wildfly/extension/undertow/deployment/UndertowDeploymentInfoService.java]: Just a note on this: In EAP 6.4 this was not configurable behavior, it was always on. So I agree it needs to be configurable, but adding it non-configurable is a true restoration of previous behavior, not just a partial one.

The basic question here is whether having it always on in 10.1 is acceptable (e.g. perf impact or something) or whether this should just wait for 11 and be done with the config option.

Comment (stuartwdouglas) [inline undertow/src/main/java/org/wildfly/extension/undertow/deployment/UndertowDeploymentInfoService.java]: Thinking about it I am ok with this going in. If it causes problems for anyone they can use a configurator to work around them programmatically, so it should be fine.

**Comment by bstansberry** [comment]
Per Stuart's line comment above, this is ok for 10.1.

**Pull request merged**